File: clients/java/java-httpserver/src/main/java/xcarpaccio/Order.java
package xcarpaccio;

import java.util.Arrays;

public class Order {
	public Double[] prices;
	public Double[] quantities;
	public String country;
	public String reduction;

	@Override
	public String toString() {
		final StringBuilder sb = new StringBuilder("Order{");
		sb.append("prices=").append(Arrays.toString(prices));
		sb.append(", quantities=").append(Arrays.toString(quantities));
		sb.append(", country='").append(country).append('\'');
		sb.append(", reduction='").append(reduction).append('\'');
		sb.append('}');
		return sb.toString();
	}
}

File: clients/cpp/OrderManagement/sources/extreme_carpaccio/order_management/Order.cpp
#include <extreme_carpaccio/order_management/Order.hpp>

#include <cstdlib>

namespace extreme_carpaccio {
namespace order_management {

template <class T>
static std::ostream& operator<< (std::ostream& stream, const std::vector<T>& v)
{
   stream << "{";
   for (std::size_t i = 0; i < v.size(); ++i)
   {
      if (i > 0)
      {
         stream << ",";
      }
      stream << v[i];
   }
   stream << '}';
   return stream;
}

std::ostream& operator<< (std::ostream& stream, const Order& order)
{
	stream << "Order{" <<
		"prices=" << order.prices <<
		", quantities=" << order.quantities <<
		", country='" << order.country + '\'' <<
		", reduction='" << order.reduction << '\'' <<
		'}';
	return stream;
}

} // namespace order_management
} // namespace extreme_carpaccio
